library/Notifications/Hook/ObjectsRendererHook.php

```diff
@@ -0,0 +1,179 @@
+ */
+abstract class ObjectsRendererHook
+{
+    /**
+     * Array of Object ID tags for each source
+     *
+     * It has the following structure : ['object source type' => ['object ID' => [object ID tags]]].
+     *
+     * @var array<string, array<string, array<string, string>>>
+     */
+    protected static $objectIdTags = [];
+
+    /**
+     * Array of HTMLs for objects with their corresponding object IDs as keys
+     *
+     * It has the following structure : ['object ID' => registered HTML for the object name].
+     *
+     * @var array<string, ValidHtml>
+     */
+    protected static $objectNameHtmls = [];
+
+    /**
+     * Get the HTML for the object names for the objects using the object ID tags
+     *
+     * @param array<string, array<string, string>> $objectIdTags Array of object ID tags with their corresponding
+     *                                                           object IDs as keys. It has the following structure:
+     *                                                           ['object ID' => [object ID tags]]
+     *
+     * @return array<string, ValidHtml> Array of HTMLs for object names with their object IDs as keys
+     */
+    abstract public function getHtmlForObjectNames(array $objectIdTags): array;
+
+    /**
+     * Get the source type of the objects
+     *
+     * @return string
+     */
+    abstract public function getSourceType(): string;
+
+    /**
+     * Create the object link for the given object ID tag
+     *
+     * @param array<string, string> $objectIdTag
+     *
+     * @return ValidHtml
+     */
+    abstract public function createObjectLink(array $objectIdTag): ValidHtml;
+
+    /**
+     * Register object ID tags to the cache
+     *
+     * @param Objects $obj
+     *
+     * @return void
+     */
+    final public static function register(Objects $obj): void
+    {
+        self::$objectIdTags[$obj->source->type][$obj->id] = $obj->id_tags;
+    }
+
+    /**
+     * Load HTMLs to be rendered for the object names to the cache using the cached objects
+     *
+     * @return void
+     */
+    final public static function load(): void
+    {
+        self::prepare(self::$objectIdTags);
+
+        self::$objectIdTags = [];
+    }
+
+    /**
+     * Prepare the objects to be rendered using the given object ID tags for each source
+     *
+     * The supplied object ID tags must have the following structure:
+     * ['object source type' => ['object ID' => [object ID tags]]].
+     *
+     * @param array<string, array<string, array<string, string>>> $objectIdTags Array of object ID tags for each source
+     *
+     * @return void
+     */
+    private static function prepare(array $objectIdTags): void
+    {
+        $objectDisplayNames = [];
+        foreach (Hook::all('Notifications\\ObjectsRenderer') as $hook) {
+            /** @var self $hook */
+            try {
+                $objectDisplayNames[] = $hook->getHtmlForObjectNames($objectIdTags[$hook->getSourceType()]);
```

Comment:
The call `$objectIdTags[$hook->getSourceType()]` must be fixed. It could throw `undefined array key` if there is no `icingadb` hook, for example.